Assets/Scripts/Duncan.cs

```diff
@@ -6,14 +6,15 @@ public class Duncan : MonoBehaviour
 {
     public float speed = 4;
 
-    public void Init(Vector2 displacement){
-        GetComponent<Rigidbody2D>().velocity = displacement * (this.speed / displacement.magnitude);
+    public Duncan Init(Vector2 displacement){
+        this.GetComponent<Rigidbody2D>().velocity = displacement * (this.speed / displacement.magnitude);
+        return this;
     }
-    void OnCollisionEnter2D(Collision2D collision)
+    protected void OnCollisionEnter2D(Collision2D collision)
     {
         ContactPoint2D contact = collision.contacts[0];
         if(contact.collider.gameObject.tag == "Wall"){
-            Destroy(this.gameObject);
+            Object.Destroy(this.gameObject);
```

Comment:
Would it make sense to make exceptions for cases where the method/variable is very reserved and usually unambiguous? For exmaple, `Object.Destroy` isn't really competing with any other `Destroy` function or method (nor should it) so I think it would make sense to use it implicitly. It lessens how much we have to write and it parallels the syntax of a built-in function

Reply:
I'd personally rather **_always_** define the origin of a static method or **_never_** define the origin of a static method, unless you have a very precise rule for when to do so or when not to.

That being said, I also have a firm preference for **_always_** doing so because it establishes a very consistent, easy to remember style rule that lowers how much you have to think when writing, reviewing, and looking back over code. If I'm looking over code and I see a function without an a defined origin, every single time I'm going to have to run the process of "okay, is this a method to tied to `this` instance that's unlabeled, or is it a static method tied to some random class that _shouldn't_ have its origin defined.

It's not worth the mental effort of having to go through that process every time I see a function without a clearly defined origin, I'd rather just type "Object." a few extra times.

Reply:
> I'd personally rather **_always_** define the origin of a static method or **_never_** define the origin of a static method, unless you have a very precise rule for when to do so or when not to.

A good rule would be "Always define the origin of a static method in our library code, including those inherited from superclasses; use implicit access to builtin functions and those exposed by Unity in its default namespaces". For reference, the default namespaces are:
```
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
```

It is similar to how folks handle builtins and default namespace for other languages: something like `max` is a builtin in Python, but you _can_ write your own `max` function and scope it so that writing `max` uses that custom function. It is a bad practice and should be avoided. You _can_ explicitly use builtin functions like `builtins.max()` and, sure, that may make it clearer when there's a bunch of other local `max` functions, but the point is that **there shouldn't be**.

Reply:
That rule is precise enough to be worth considering.

What value is added by the rule exactly? Every rule we add to the MUGA increases the complexity of our coding style, adds more things we have to consciously think about while writing code to make sure we're actively adhering to, so I want to make sure that each rule actually does add value in some form. Adding an underscore to everything private or protected adds value by clarifying the non-public access of the variable or function just from its name. What additional value is added by having writers consciously thinking about adhering to this rule?

If all that's added is freeing a coder from having to type `Object.` or `GameObject.` in front of some static methods within the default namespace, that doesn't seem worth it to me, it's a very small inconvenience not significantly worse than remembering to prepend an underscore to private methods. The information communicated by seeing `Object.` or `GameObject.` before a method name is approximately the same as seeing a method without a prepended class name (telling the reader that it's a static method from a class that happens to be in the default namespace vs. that it's a static method from the default namespace). So we add complexity in how we treat static methods by dividing them into two groups that we handle differently, and in return we get... what exactly?

Reply:
Honestly mainly to stay more aligned with _other_ style conventions. At this point it may be subjective, since it'll be cognitive load for me to _deviate_ from that, while it would be load for you to _conform_ to that. We could just call it a split vote and let @JosriqueJotaro tie-break for us?

Reply:
Yeah, that sounds super fair. Split vote. @JosriqueJotaro pick whatever you'd personally prefer of the two choices. If the majority of the team prefers one, we should do that one.

Reply:
I apologize for my slow response - I was very busy this weekend and didn't have time Friday to respond. It depends on how long the origins for static functions will be. As the project becomes more complex, will our origins become significantly longer? If the answer is yes, then we adopt the rule, and if not, we don't.

Reply:
> I apologize for my slow response - I was very busy this weekend and didn't have time Friday to respond. It depends on how long the origins for static functions will be. As the project becomes more complex, will our origins become significantly longer? If the answer is yes, then we adopt the rule, and if not, we don't.

Hmmm... well, if we wanted to call a static function without prepending the origin, we would need to include whatever namespace the function is under, which I think would also make it callable from the class it's contained it since the class would likely belong to the same namespace, right @Micky774 ?

Reply:
@JosriqueJotaro

Reply:
@JosriqueJotaro

Reply:
@Nick-Angel